### DISCUSSION TITLE
Backup & restore

### DISCUSSION THREAD

**Discussion by jens-ylja** [ytyou/ticktock#39] [General]
Hi,
from the discussion regarding data export I deduct that simply copying all data files (and config) should do the job.

But there are some details:
1. I think the ticktock instance should be stopped before taking the backup to get a consistent state. Right?
2. How to trigger a graceful shutdown of the ticktock instance - including flush of all data to disk? Would a SIGTERM do the job?

Thanks

**Comment by ytyou** [comment]
1. Correct;
2. SIGINT should do, or just run admin/stop.sh, which post an HTTP request to TickTockDB.

**Reply by jens-ylja** [reply]
OK, I've overseen the POST to api/admin?cmd=stop -> quite the best solution.

**Comment by ylin30** [comment]
@jens-ylja FYI, please wait for "Shutdown process complete" message to continue. See below example run:

```
ylin30@orangepizero2:~/ticktock$  TickTockDB v0.11.5,  Maintained by
 Yongtao You (yongtao.you@gmail.com) and Yi Lin (ylin30@gmail.com).
 This program comes with ABSOLUTELY NO WARRANTY. It is free software,
 and you are welcome to redistribute it under certain conditions.
 For details, see <https://www.gnu.org/licenses/>.
Writing to log file: /home/ylin30/ticktock/log/ticktock.log

ylin30@orangepizero2:~/ticktock$ ./admin/stop.sh
Exiting now.
ylin30@orangepizero2:~/ticktock$ Start shutdown process...
Shutdown process complete

[1]+  Done                    ./bin/tt -c conf/tt.conf --tsdb.timestamp.resolution millisecond
ylin30@orangepizero2:~/ticktock$
```
